dolphinscheduler-api/src/main/java/org/apache/dolphinscheduler/api/audit/AuditPublishService.java

```diff
@@ -0,0 +1,87 @@
+package org.apache.dolphinscheduler.api.audit;
+
+import org.apache.dolphinscheduler.api.configuration.AuditConfiguration;
+import org.apache.dolphinscheduler.common.enums.AuditModuleType;
+import org.apache.dolphinscheduler.common.enums.AuditOperationType;
+import org.apache.dolphinscheduler.dao.entity.User;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Component;
+
+import javax.annotation.PostConstruct;
+import java.util.Date;
+import java.util.List;
+import java.util.concurrent.BlockingQueue;
+import java.util.concurrent.LinkedBlockingQueue;
+
+@Component
+public class AuditPublishService {
+
+    /**
+     * audit message queue
+     */
+    private BlockingQueue<AuditMessage> queue = new LinkedBlockingQueue<>(1000);
+
+    /**
+     * subscribers list
+     */
+    @Autowired
+    private List<AuditSubscriber> subscribers;
+
+    @Autowired
+    AuditConfiguration auditConfiguration;
+
+    private static final Logger logger = LoggerFactory.getLogger(AuditPublishService.class);
+
+    /**
+     * create a daemon thread to process the message queue
+     */
+    @PostConstruct
+    private void init() {
+        Thread thread = new Thread(() -> doPublish());
+        thread.setDaemon(true);
+        thread.start();
+    }
+
+    /**
+     * publish a new audit message
+     *
+     * @param message audit message
+     */
+    public void publish(AuditMessage message) {
+        queue.offer(message);
+    }
+
+    /**
+     *  subscribers execute the message processor method
+     */
+    private void doPublish() {
+        AuditMessage message;
+        if (auditConfiguration.isAuditGlobalControlSwitch()) {
```

Comment:
There might be a bug, if the switch set to false, the application will still add `AuditMessage` to the queue.

It is better to move this switch to `publish` method and `init` method.